pallets/loans/src/rate_model.rs

```diff
@@ -165,19 +185,57 @@ impl CurveModel {
     pub fn new_model(base_rate: Rate) -> CurveModel {
         Self { base_rate }
     }
+}
 
-    /// Check the curve model for sanity
-    pub fn check_model(&self) -> bool {
-        true
+impl InterestRateModel for CurveModel {
+    fn check_model(&self) -> bool {
+        false
     }
 
-    /// Calculates the borrow interest rate of curve model
-    pub fn get_borrow_rate(&self, _utilization: Ratio) -> Option<Rate> {
-        // TODO: Need to implement the curve model
+    fn get_borrow_rate(&self, _: Ratio) -> Option<Rate> {
         None
     }
 }
 
+#[derive(Clone, Copy, Decode, Default, Encode, Eq, PartialEq, RuntimeDebug)]
+pub struct NewCurveNameModel {
+    borrow_rate: Rate,
+    reserve_ratio: Ratio,
+    staking_apy: Ratio,
+}
+
+impl NewCurveNameModel {
+    pub fn new(borrow_rate: Rate, reserve_ratio: Ratio, staking_apy: Ratio) -> Self {
+        Self {
+            borrow_rate,
+            reserve_ratio,
+            staking_apy,
```

Comment:
As described on #229, `staking_apy` should be `xKSM_exchange_rate / xKSM_exchange_rate(last_week) * 52.1429` but it is currently a percentage because the semantic of `52.1429` is unknown. Or maybe it is just me that couldn't spot its meaning.
Please, feel free to discuss which direction should `staking_apy` take.

Reply:
staking_apy should come from liquid_staking pallet, and it has not been implemented yet. We can just use the const 14% for now.

Reply:
I agree. We can use a const for now but highly suggest that we can have an interface to export the current staking_apy

Reply:
Changed `staking_apy` to a constant

Reply:
The interest rate is dynamic, it's related to the staking apy at product level, not the code/protocol level.
Make it configurable from runtime makes more sense.

Reply:
IIUC, `interest_rate == borrow_rate` parameter retrieved from a runtime configuration, therefore, a new `Get<Rate>` type was added to `loans::Config`.